osidb/migrations/0123_alert_refactor.py

```diff
@@ -0,0 +1,153 @@
+"""
+Written manually on 2024-03-26
+
+Create new Alert model, convert alerts stored in a JSON field to Alert records,
+and remove the JSON field.
+"""
+
+from django.conf import settings
+from django.db import migrations, models
+import django.db.models.deletion
+from itertools import islice
+import uuid
+
+from osidb.core import set_user_acls
+
+
+BATCH_SIZE = 1000
+
+# From a migration we cannot get the models that implement the ALertMixin programmatically so
+# these need to be declared explicitly.
+ALERTABLE_MODELS = [
+    "Flaw",
+    "Affect",
+    "Snippet",
+    "FlawCVSS",
+    "AffectCVSS",
+    "Tracker",
+    "FlawMeta",
+    "FlawComment",
+    "FlawAcknowledgment",
+    "FlawReference",
+    "Package",
+]
+
+
+def generate_alertmixins(apps, model):
+    Alert = apps.get_model("osidb", "Alert")
+    # Convert alerts in JSON format to Alert records.
+    # At this point in the migration we have both the old _alerts field and the
+    # new alerts relation.
+    for instance in (
+        model.objects.exclude(_alerts={}).exclude(_alerts__isnull=True).iterator()
+    ):
+        for key in instance._alerts.keys():
+            values = instance._alerts.get(key)
+            yield Alert(
+                name=key,
+                description=values.get("description", ""),
+                alert_type=values.get("type", "warning"),
+                resolution_steps=values.get("resolution_steps", ""),
+                content_object=instance,
```

Comment:
Cannot apply the migration, I am getting prompted with a traceback on this particular line:
```
...
TypeError: Alert() got an unexpected keyword argument 'content_object'
```

Reply:
@dmonzonis So this was actually some problem on my side, I issued `make compose-down` and build everything from scratch and it works.